src/client/components/equalizer/equalizer.js

```diff
@@ -0,0 +1,200 @@
+    heightArray: undefined
+  }
+
+  componentDidMount() {
+    let canvasEl = document.querySelector("canvas").getContext("2d");
+    let numPoints = this.state.analyser.frequencyBinCount - 80;
+    let uint8Array = new Uint8Array(numPoints);
+    this.setState({
+      ctx: canvasEl,
+      numPoints: numPoints,
+      heightArray: uint8Array
+    });
+    this.detectStreamSoundFromMicrophon();
+  }
+
+  createSoundStream = (stream) => {
+    let {audioContext: context} = this.props.audioData;
+    let audioLineIn = new Audio();
+
+    audioLineIn.srcObject = stream;
+    audioLineIn.muted = true;
+    //hark - JS module that listens to an audio stream, and emits events indicating whether the user is speaking or not 
+    let options = {};
+    let speechEvents = Hark(stream, options);
+    let htmlinfo = document.getElementById("stream_detecting");
+    speechEvents.on('speaking', () => {
+      htmlinfo.innerHTML = `speaking`;
+    });
+    speechEvents.on('stopped_speaking', () => {
+      htmlinfo.innerHTML = `no stream detekting`;
+    });
+    let sourceStream = context.createMediaStreamSource(stream);
+    this.props.createStreamData({
+      audioLineIn: audioLineIn,
+      sourceStream
+    });
+  }
+
+  detectStreamSoundFromMicrophon = () => {
+    if (navigator.mediaDevices) {
+      navigator.mediaDevices.getUserMedia({
+          audio: true
+        })
+        .then(stream => this.createSoundStream(stream))
+        .catch((e) => {
+          throw new Error(e);
+        });
+    } else {
+      throw new Error('browser doesnt support audio API');
+    }
+  }
+
+  widthMerge = (e) => {
+    this.props.mergeCanvasWidth(e);
+  }
+
+  playSoundFromFile = () => {
+    let {audioFromFile: soundFromFile, playPauseState} = this.props.audioData;
+    if (!playPauseState) {
+      soundFromFile.play();
+      this.equaliserRun();
+    } else {
+      soundFromFile.pause();
+    }
+    this.props.playPauseSoundFromFile();
+  }
+
+  startMuteStream = () => {
+    let {
+      audioStream,
+      audioContext,
+      analyser,
+      streamSource,
+      startMuteState
+    } = this.props.audioData;
+
+    if (!startMuteState) {
+      streamSource.connect(analyser);
+      analyser.connect(audioContext.destination);
+      //play/pause function doesn't work 
+      audioStream.play();
+      this.equaliserRun();
+      this.props.startMuteStreamAudio();
+    } else {
+      streamSource.disconnect(analyser);
+      audioStream.pause();
+      this.props.startMuteStreamAudio();
+    }
+  }
+
+  equaliserRun = () => {
+    requestAnimationFrame(this.renderEqualizer);
+  }
+
+  renderEqualizer = () => {
+    let {analyser, heightArray, ctx, numPoints} = this.state;
+    let flagColor = true;
+    analyser.getByteFrequencyData(heightArray);
+    let width = ctx.canvas.width;
+    let height = ctx.canvas.height;
+    let countcolumns = Math.floor(ctx.canvas.width / 52);
+    let columnwidth = Math.floor(5 / 6 * ctx.canvas.width / 52);
```

Comment:
please do not use magic numbers like 5, 6 or 52, but use constants with readable and understandable names instead

Reply:
done